### ISSUE TITLE
Type error with latest `rescript-compiler#master`

### ISSUE THREAD

**Issue by mattdamon108** [zth/rescript-relay#391]
Hi, I think this issue is based on the unstable compiler version, but I want to have a clue to solve to proceed the tests with compiler and JSX v4 on my company project. My company projects are using heavily the relay with rescript-relay.
I found some error.
```rescript
// RescriptRelay.res

let queryRefToPromise = token => {
    Js.Promise.make((~resolve, ~reject as _) => {
      switch token->queryRefToObservable {
      | None => resolve(. Error())
      | Some(o) =>
        let _: Observable.subscription = o->{
          open Observable
          subscribe(makeObserver(~complete=() => resolve(. Ok()), ()))
        }
      }
    })
  }
```
```
  We've found a bug for you!
  /Users/woonki/Github/works/gl/sources/sinsunhi-frontend-mirror/node_modules/rescript-relay/src/RescriptRelay.res:767:21-69

  765 ┆   let _: Observable.subscription = o->{
  766 ┆     open Observable
  767 ┆     subscribe(makeObserver(~complete=() => resolve(. Ok()), ()))
  768 ┆   }
  769 ┆ }

  This has type: Observable.observer<'a>
  Somewhere wanted: Observable.t<'b>
```
I think I can fix it as below
```rescript
         let _: Observable.subscription = {
          open Observable
          subscribe(o, makeObserver(~complete=() => resolve(. Ok()), ()))
        }
```
But, I face many build error from different places in my projects.

<img width="812" alt="image" src="https://user-images.githubusercontent.com/36963424/191062406-aac045a1-0b6c-4b32-81b3-71277a0b7ca9.png">

Can I get a tip to fix the build error?

**Comment by zth** [comment]
This is actually already fixed in https://github.com/zth/rescript-relay/commit/fbc78b86d57c155d87db33f3f784b695c4342c60 but not released yet. I'll make a rc.5 tonight with the fix.

**Comment by mattdamon108** [comment]
> This is actually already fixed in [fbc78b8](https://github.com/zth/rescript-relay/commit/fbc78b86d57c155d87db33f3f784b695c4342c60) but not released yet. I'll make a rc.5 tonight with the fix.

Lovely! Thank you! 👍

**Comment by zth** [comment]
@mattdamon108 `1.0.0-rc.5` is out now.
